controller/linodemachine_controller.go

```diff
@@ -317,30 +320,40 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 		linodeInstance = &linodeInstances[0]
 	case 0:
 		// get the bootstrap data for the Linode instance and set it for create config
-		createConfig, err := r.newCreateConfig(ctx, machineScope, tags, logger)
+		createOpts, err := r.newCreateConfig(ctx, machineScope, tags, logger)
 		if err != nil {
 			logger.Error(err, "Failed to create Linode machine create config")
 
 			return nil, err
 		}
 
 		if machineScope.LinodeCluster.Spec.VPCRef != nil {
-			iface, err := r.getVPCInterfaceConfig(ctx, machineScope, createConfig.Interfaces, logger)
+			iface, err := r.getVPCInterfaceConfig(ctx, machineScope, createOpts.Interfaces, logger)
 			if err != nil {
 				logger.Error(err, "Failed to get VPC interface config")
 
 				return nil, err
 			}
 
-			createConfig.Interfaces = append(createConfig.Interfaces, *iface)
+			createOpts.Interfaces = append(createOpts.Interfaces, *iface)
 		}
 
-		linodeInstance, err = machineScope.LinodeClient.CreateInstance(ctx, *createConfig)
+		linodeInstance, err = machineScope.LinodeClient.CreateInstance(ctx, *createOpts)
 		if err != nil {
 			logger.Error(err, "Failed to create Linode machine instance")
 
 			return nil, err
 		}
+
+		// create etcd disk only if this is a control plane node
+		if kutil.IsControlPlaneMachine(machineScope.Machine) {
+			if err = r.configureDisks(ctx, logger, machineScope, linodeInstance.ID); err != nil {
+				logger.Error(err, "Failed to configure instance disks")
+
+				return nil, err
+			}
```

Comment:
Maybe we should put the boot step in the kutil if loop instead of configureDisks so that we can separate if it failed to configure disk or failed to boot the instance?

Reply:
sure, sounds good